include/cudaq/Support/TargetConfig.h

```diff
@@ -97,6 +97,8 @@ struct BackendEndConfigEntry {
   SimulationBackendSetting SimulationBackend;
   /// Any conditional compile/link flags configurations.
   std::vector<ConditionalBuildConfig> ConditionalBuildConfigs;
+  // Optional entry to manually specify / override server helper
+  std::optional<std::string> ServerHelper = std::nullopt;
```

Comment:
nit: I don't think you need an initializer expression here as the default ctor will be nullopt.